array_api_strict/tests/test_array_object.py

```diff
@@ -410,6 +411,78 @@ def _matmul_array_vals():
                     x.__imatmul__(y)
 
 
+@pytest.mark.parametrize(
+    "op",
+    [
+        op for op, dtypes in binary_op_dtypes.items()
+        if dtypes not in ("real numeric", "floating-point")
+    ],
+)
+def test_binary_operators_vs_numpy_int(op):
+    """np.int64 is not a subclass of int and must be disallowed"""
+    a = asarray(1)
+    i64 = np.int64(1)
+    with pytest.raises(TypeError, match="Expected Array or Python scalar"):
+        getattr(a, op)(i64)
+
+
+@pytest.mark.parametrize(
+    "op",
+    [
+        op for op, dtypes in binary_op_dtypes.items() 
+        if dtypes not in ("integer", "integer or boolean")
+    ],
+)
+def test_binary_operators_vs_numpy_float(op):
+    """
+    np.float64 is a subclass of float and must be allowed.
+    np.float32 is not and must be rejected.
+    """
+    a = asarray(1.)
+    f64 = np.float64(1.)
+    f32 = np.float32(1.)
+    func = getattr(a, op)
+    for op in binary_op_dtypes:
+        assert isinstance(func(f64), Array)
+        with pytest.raises(TypeError, match="Expected Array or Python scalar"):
+            func(f32)
```

Comment:
This test also reflects the status quo: the behavior is exactly the same on main.